[BUG] Flaky test_compare_template_stack_dsk_bf16

**Describe the bug**

`openfold3/tests/test_kernels.py::TestKernels::test_compare_template_stack_dsk_bf16` fails intermittently (~2% of runs) on NVIDIA GB10 (Grace Blackwell). The threshold is the outlier: this test hardcodes `eps=2e-2` for the max-abs-diff comparison, while peer bf16 kernel-comparison tests use `4e-2`:

- `test_compare_pairformer_dsk_bf16` → `4e-2`
- `test_compare_diffusion_transformer_dsk_bf16` → `4e-2`
- `test_compare_template_stack_dsk_bf16` → **`2e-2`** (this one)

Every observed failure produces `max_abs_diff = 0.0234375` (= 3/128, a single bf16 ULP step at output magnitude ~1), only 17% over the threshold. Underlying DS4S vs native attention agreement is fine — in fp32 the kernels agree to ~1e-3 (see investigation summary below).

This is very rare but i have had at least one user report this to me. 

**To Reproduce**

```
pixi run -e openfold3-cuda12-pypi \
  python -m pytest openfold3/tests/test_kernels.py::TestKernels::test_compare_template_stack_dsk_bf16
```

Single runs pass most of the time. To observe the flakiness, loop it:

```
for i in $(seq 1 100); do
  pixi run -e openfold3-cuda12-pypi \
    python -m pytest openfold3/tests/test_kernels.py::TestKernels::test_compare_template_stack_dsk_bf16 -q -x >/dev/null 2>&1 \
    && echo "$i PASS" || echo "$i FAIL"
done
```

This is UnitTest test (legacy), rather than pytest so we can't just run with --count. 

Observed: ~2/100 fail. No special query / runner config — the test is fully self-contained.

**Expected behavior**

The test passes deterministically, or fails with a clear correctness signal rather than a tolerance grazing-the-tail flake.

**Stack trace**

```
openfold3/tests/test_kernels.py:802: in test_compare_template_stack_dsk_bf16
    self._compare_template_stack(use_deepspeed_evo_attention=True, dtype=torch.bfloat16)
openfold3/tests/test_kernels.py:791: in _compare_template_stack
    compare_utils.assert_max_abs_diff_small(out_repro, out_repro_ds, 2e-2)
openfold3/tests/compare_utils.py:73: in assert_max_abs_diff_small
    _assert_abs_diff_small_base(torch.max, expected, actual, eps)
E   AssertionError: Scalars are not close!
E   Expected 0.0 but got 0.0234375.
E   Absolute difference: 0.0234375 (up to 0.02 allowed)
E   Relative difference: inf (up to 0.016 allowed)
```

**Configuration**

- OS: Ubuntu 24.04.4 LTS
- GPU: NVIDIA GB10 (Grace Blackwell), ~131 GB unified memory
- Python: 3.14.5
- PyTorch: 2.11.0+cu129
- CUDA: 12.9
- Environment: `openfold3-cuda12-pypi` (pixi); also reproduces under `openfold3-cuda13-pypi`
- Installation: built from repo via `pixi install`

**Additional context**

Investigated by running a measurement harness 200×–unseeded and 20×-seeded. Summary:

| Experiment | Observation |
|---|---|
| DSK bf16, N=200, unseeded | p50=0.0156, p99=0.0234, max=0.0234; **2.0% over 2e-2, 0% over 4e-2**; only 5 distinct ULP-class values |
| DSK fp32, N=50, unseeded | max=0.002; DS4S agrees with native attention to ~1e-3 → **no kernel correctness gap** |
| DSK bf16, N=20, seeded (`manual_seed`+`cudnn.deterministic=True`) | 3 distinct values across 20 runs → DS4S has minor reduction-order nondeterminism on Blackwell that standard `cudnn` flags don't cover. Bounded and benign, but worth flagging separately. |

Suggested minimal fix: align this test with its peers by using `eps=4e-2` for the bf16 case at [openfold3/tests/test_kernels.py:791](openfold3/tests/test_kernels.py#L791):

```python
eps = 4e-2 if dtype == torch.bfloat16 else 2e-2
compare_utils.assert_max_abs_diff_small(out_repro, out_repro_ds, eps)
```

Projected effect: 0/200 failures, fp32 path unchanged.

--- a/openfold3/tests/test_kernels.py
+++ b/openfold3/tests/test_kernels.py
@@ -716,6 +716,7 @@ class TestKernels(unittest.TestCase):
         use_triton_triangle_kernels=False,
         dtype=torch.float32,
         chunk_size=None,
+        eps=2e-2,
     ):
         """
         Compare Template Stack output with and without using different optimized
@@ -788,7 +789,7 @@ class TestKernels(unittest.TestCase):
                 use_triton_triangle_kernels=use_triton_triangle_kernels,
             )
 
-            compare_utils.assert_max_abs_diff_small(out_repro, out_repro_ds, 2e-2)
+            compare_utils.assert_max_abs_diff_small(out_repro, out_repro_ds, eps)
 
     @compare_utils.skip_unless_ds4s_installed()
     def test_compare_template_stack_dsk_fp32(self):
@@ -802,6 +803,7 @@ class TestKernels(unittest.TestCase):
         self._compare_template_stack(
             use_deepspeed_evo_attention=True,
             dtype=torch.bfloat16,
+            eps=4e-2,
         )
 
     @compare_utils.skip_unless_ds4s_installed()

